XXX-buffer-alloc-from.md

```diff
@@ -0,0 +1,50 @@
+# Buffer.alloc / Buffer.allocUnsafe / Buffer.from
+
+Status | Draft
+------ | ----
+Author | James M Snell
+Date   | 25 Jan 2016
+
+## Purpose
+
+* Soft-Deprecate the existing `Buffer()` constructors and introduce new
+  `Buffer.from()`, `Buffer.alloc()` and `Buffer.allocUnsafe()` factory methods
+  to address Buffer API usability and security concerns.
+* Add `--zero-fill-buffers` command line option to node to force all Buffers
+  to zero-fill when created
+
+## Details
+
+The details of the proposal are captured by [Pull Request 4682][] in the
+nodejs/node repository (https://github.com/jasnell/node/tree/buffer-safe).
+
+### Buffer.alloc(size[, fill])
+
+Allocates a new initialized buffer of `size` bytes. If `fill` is specified
+and is a string, the Buffer is allocated off the shared pool by calling:
```

Comment:
Yep, the proposal PR has been updated to account for that. For
completeness, @trevnorris and I are also working on allowing both `fill()`
and `alloc(size, fill)` to accept a `Buffer` and to accept an `encoding`
when a string is passed in, e.g:

```
// these are silly examples, but you should get the idea:
Buffer.alloc(10, 'foo', 'ascii');
Buffer.alloc(10, Buffer.from('foo'));
Buffer.alloc(10).fill('foo', 'ascii');
Buffer.alloc(10).fill(Buffer.from('foo'));
```

On Thu, Jan 28, 2016 at 6:50 AM, Feross Aboukhadijeh <
notifications@github.com> wrote:

> In XXX-buffer-alloc-from.md
> <https://github.com/nodejs/node-eps/pull/7#discussion_r51131216>:
>
> > +
> > +* Soft-Deprecate the existing `Buffer()` constructors and introduce new
> > +  `Buffer.from()`, `Buffer.alloc()` and `Buffer.allocUnsafe()` factory methods
> > +  to address Buffer API usability and security concerns.
> > +* Add `--zero-fill-buffers` command line option to node to force all Buffers
> > +  to zero-fill when created
> > +
> > +## Details
> > +
> > +The details of the proposal are captured by [Pull Request 4682][] in the
> > +nodejs/node repository (https://github.com/jasnell/node/tree/buffer-safe).
> > +
> > +### Buffer.alloc(size[, fill])
> > +
> > +Allocates a new initialized buffer of `size` bytes. If `fill` is specified
> > +and is a string, the Buffer is allocated off the shared pool by calling:
>
> Just want to echo @ChALkeR <https://github.com/ChALkeR>'s point that
> buf.fill(val) currently accepts number values and this API should do that
> too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node-eps/pull/7/files#r51131216>.
>